File: templates/hexagonal-spring-boot/rest-adapter/src/main/java/packageName/rest/exception/ExampleExceptionHandler.java
package packageName.rest.exception;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import org.springframework.web.context.request.ServletWebRequest;
import org.springframework.web.context.request.WebRequest;
import packageName.domain.exception.ExampleNotFoundException;

@RestControllerAdvice(basePackages = {"packageName"})
public class ExampleExceptionHandler {

  @ExceptionHandler(value = ExampleNotFoundException.class)
  public final ResponseEntity<ExampleExceptionResponse> handleExampleNotFoundException(final WebRequest request) {
    return ResponseEntity.status(HttpStatus.NOT_FOUND).body(ExampleExceptionResponse.builder().message("Example not found").path(((ServletWebRequest) request).getRequest().getRequestURI()).build());
  }
}

File: templates/hexagonal-spring-boot-kotlin/rest-adapter/src/main/kotlin/packageName/rest/exception/ExampleExceptionHandler.kt
package packageName.rest.exception

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice
import org.springframework.web.context.request.ServletWebRequest
import org.springframework.web.context.request.WebRequest
import packageName.domain.exception.ExampleNotFoundException

@RestControllerAdvice(basePackages = ["packageName"])
class ExampleExceptionHandler {

    @ExceptionHandler(value = [ExampleNotFoundException::class])
    fun handleExampleNotFoundException(request: WebRequest): ResponseEntity<ExampleExceptionResponse> {
        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(ExampleExceptionResponse("Example not found", (request as ServletWebRequest).request.requestURI))
    }
}
